__tests__/integration/tokenAssigned.spec.js

```diff
@@ -0,0 +1,53 @@
+const { publish, subscribe} = require("../../server/infra/messaging/rabbit-mq-messaging");
+const knex = require("../../server/infra/database/knex");
+const {expect} = require("chai");
+const registerEventHandlers = require('../../server/services/event-handlers.js');
+const { unsubscribeAll } = require("../../server/infra/messaging/rabbit-mq-messaging");
+const {v4} = require("uuid");
+
+describe("tokenAssigned", () => {
+
+  beforeEach(async () => {
+    //load server
+    registerEventHandlers();
+    await knex("capture_feature").del();
+  });
+
+  afterEach(async () => {
+    await unsubscribeAll();
+  });
+
+  it("Successfully handle tokenAssigned event", async () => {
+    const capture_id = "63e00bca-8eb0-11eb-8dcd-0242ac130003";
+    const token_id = "9d7abad8-8eb0-11eb-8dcd-0242ac130003";
+    const capture = {
+      id: capture_id,
+      lat: 0,
+      lon: 0,
+      location: `POINT(0 0)`,
+      field_user_id: 0,
+      field_username: "fake_name",
+      token_id,
+      wallet_name: "oldone",
+      capture_taken_at: new Date(),
+      created_at: new Date(),
+      updated_at: new Date(),
+    }
+    //prepare the capture before the wallet event
+    await knex("capture_feature").insert(capture);
+    const wallet_name_new = "newone";
+    const message = {
+      "type": "TokensAssigned",
+      "wallet_name": wallet_name_new,
+      "entries": [
+        { "capture_id": capture_id, "token_id": token_id },
+      ],
+    }
+    publish("token-assigned", undefined, message, (e) => console.log("result:", e));
```

Comment:
could we possibly skip involving the queue  and instead use the registered handler for the event to test it? 
`publish` function is meant for dispatching messages from the originating system. `token-assigned` here is a consumed message and should enter the web-map-query-consumer via `subscribe` function.  I am not sure if this test is really exercising/simulating the flow of message into web-map-query-consumer (`publish` sends out messages to a topic to be consumed by another service and not by itself). I recommend to skip the queues so it is simpler.

Reply:
Arun the whole test passed, it proved that the queue and DB process works well, so it's useful, the argument point I guess is that if we worth the effort doing this. 
I prefer to link them all together and test them as a whole, it makes me more confident on the code. If we skip it and call the handler directly, it's more like a unit test, the queue workflow is uncovered.
From another angle, skipped the queue workflow, I still need to test it by myself, by operating the queue manually, it's more inefficient IMO, and another bonus of building this test is that I got a test to cover my code in the future.
Maintaining the test is a question, but a higher level integration test is easier to maintain than a lower level of unit tests, I think.